### PR TITLE
Adding transfer learning content and documentation around it

### PR DESCRIPTION
**Pull request by yradsmikham**

### CODE REVIEW

File: microsoft/transfer.py [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+                         "How many passes to make over the dataset before "
+                         "halting training.")
+    flags.DEFINE_integer("max_steps", None,
+                         "The maximum number of iterations of the training loop.")
+    flags.DEFINE_integer("export_model_steps", 1000,
+                         "The period, in number of steps, with which the model "
+                         "is exported for batch prediction.")
+
+    # Other flags.
+    flags.DEFINE_string("base_checkpoint", "model.ckpt-3425",
```
Comment (evanlouie) [inline microsoft/transfer.py]: Any particular reason it defaults to `model.ckpt-3425`?

File: microsoft/transfer.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+                        "pretrained checkpoint")
+    flags.DEFINE_integer("num_readers", 8,
```
Comment (evanlouie) [inline microsoft/transfer.py]: Change to default to `multiprocessing.cpu_count()`

File: microsoft/transfer.py [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Binary for training Tensorflow models on the YouTube-8M dataset."""
+
+import json
+import os
+import time
+import sys
+
+sys.path.insert(0, 'youtube_8m')
```
Comment (evanlouie) [inline microsoft/transfer.py]: Hacky. We'll fix this down the line to use standard imports once we reorg the modules

**Comment by evanlouie** [comment]
also, rebase into single commit of possible

**Comment by evanlouie** [comment]
fixes #5 #6 #8

File: microsoft/transfer.py (hunk 1 above)
Comment (yradsmikham) [reply microsoft/transfer.py]: No reason, this was left as default from Mihaela. Should I change it to model.ckpt-0?

Comment (evanlouie) [reply microsoft/transfer.py]: yea, ckpt-0 makes sense

**Comment by evanlouie** [comment]
closes #5 #6 #8

**Pull request merged**